paddle/fluid/operators/conv_transpose_mkldnn_op.cc

```diff
@@ -0,0 +1,299 @@
+/* Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License. */
+
+#include "paddle/fluid/framework/data_layout_transform.h"
+#include "paddle/fluid/framework/op_registry.h"
+#include "paddle/fluid/memory/malloc.h"
+#include "paddle/fluid/platform/mkldnn_reuse.h"
+
+namespace paddle {
+namespace operators {
+
+using Tensor = framework::Tensor;
+using framework::DataLayout;
+
+template <typename T>
+class ConvTransposeMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
+ public:
+  void Compute(const paddle::framework::ExecutionContext& ctx) const override {
+    PADDLE_ENFORCE(paddle::platform::is_cpu_place(ctx.GetPlace()),
+                   "It must use CPUPlace.");
+
+    const bool is_test = ctx.Attr<bool>("is_test");
+    PADDLE_ENFORCE(
+        is_test == true,
+        "ConvTransposeMKLDNN works only for inference!. Set is_test = True");
```

Comment:
You should update
https://github.com/PaddlePaddle/Paddle/blob/6224e61fd94e6ad87f18c2808a76256b516fa3f3/paddle/fluid/framework/ir/is_test_pass.cc#L27-L37
to make `is_test_pass` work on `ConvTransposeMKLDNN `

Reply:
I don't think we need update op_list.  The "conv_transpose" OP already have attribute "[is_test](https://github.com/PaddlePaddle/Paddle/blob/06d8777ec2857020da34b60ba359cf9b8e963b4e/paddle/fluid/operators/conv_transpose_op.cc#L113)". According to below code snippet, "is_set" should be already set to true.
`      if (op->HasAttr("is_test")) {
        op->SetAttr("is_test", true);
      } else if (std::find(begin(op_list), end(op_list), op->Type()) !=
                 end(op_list)) {
        op->MutableAttrMap()->insert(
            std::pair<std::string, Attribute>("is_test", true));
      }`

Reply:
@jianhang-liu You are right, but why this PR does not increase the speed?

Reply:
I wonder ConvTransposeMKLDNNOpKernel had NOT been selected (and run). In MKLDNN log, no "deconv" primitive was displayed.

Reply:
@luotao1 Here is output from MKLDNN verbose, for UT:
...
mkldnn_verbose,exec,deconvolution,ref:any,forward_inference,fsrc:nchw fwei:nchw fbia:undef fdst:nchw,alg:deconvolution
...

Please check (printf) why Conv2d_transpose is not called